nixos/modules/services/web-apps/codeberg-pages.nix

```
@@ -0,0 +1,148 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (lib)
+    mkIf
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    ;
+  inherit (lib) types;
+
+  cfg = config.services.codeberg-pages;
+in
+{
+  meta.maintainers = with lib.maintainers; [
+    NotAShelf
+  ];
+
+  options = {
+    services.codeberg-pages = {
+      enable = mkEnableOption "Codeberg pages server";
+      package = mkPackageOption pkgs "codeberg-pages" { };
+      stateDir = mkOption {
+        type = types.path;
+        default = "/var/lib/codeberg-pages";
+        description = "Directory in which state will be stored.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "codeberg-pages";
+        description = "User account under which Codeberg Pages will run.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "codeberg-pages";
+        description = "Group under under which Codeberg Pages will run.";
+      };
+
+      settings = mkOption {
+        type = with types; submodule { freeformType = attrsOf anything; }; # TODO: pick available types more carefully?
+        default = { };
+        example = {
+          ACME_ACCEPT_TERMS = true;
+          ENABLE_HTTP_SERVER = true;
+
+          GITEA_ROOT = "git.exampledomain.tld";
+        };
+
+        description = ''
+          Configuration values to be passed to the codeberg pages server
+          as environment variables.
+          [pages-server documentation]: https://codeberg.org/Codeberg/pages-server#environment-variables
+          See [pages-server documentation] for available options.
+          For sensitive values, prefer using {option}`services.codberg-pages.environmentFile`.
+        '';
+      };
+
+      environmentFile = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        example = "/run/secret/pages-server.env";
+        description = ''
+          An environment file as defined in {manpage}`systemd.exec(5)`.
+          Sensitive information, such as {env}`GITEA_API_TOKEN`, may be passed
+          to the service without adding them to the world-readable Nix store.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd = {
+      tmpfiles.rules = [ "d '${cfg.stateDir}' 0750 ${cfg.user} ${cfg.group} - -" ];
```

Please use `tmpfiles.settings` instead.

```suggestion
      tmpfiles.settings = {
        "10-codeberge-pages" = {
          ${cfg.statedir} = {
            d = {
              inherit (cfg) group user;
              mode = "0750";
            };
          };
        };
      };
```